force-app/main/default/lwc/bulkServiceDeliveryUI/serviceDeliveryFieldSets.js

```diff
@@ -0,0 +1,118 @@
+import CONTACT_FIELD from "@salesforce/schema/ServiceDelivery__c.Contact__c";
+import SERVICE_FIELD from "@salesforce/schema/ServiceDelivery__c.Service__c";
+import QUANTITY_FIELD from "@salesforce/schema/ServiceDelivery__c.Quantity__c";
+import PROGRAM_ENGAGEMENT_FIELD from "@salesforce/schema/ServiceDelivery__c.ProgramEngagement__c";
+
+import selectService from "@salesforce/label/c.Select_Service";
+import selectEngagement from "@salesforce/label/c.Select_Program_Engagement";
+
+const SHORT_DATA_TYPES = ["DOUBLE", "INTEGER", "BOOLEAN"];
+const LONG_DATA_TYPES = ["TEXTAREA"];
+
+export class ServiceDeliveryFieldSets {
+    fieldSets = {};
+    currentFieldSetName;
+
+    constructor(data) {
+        this.setFieldSets(data);
+    }
+
+    setFieldSets(fieldSets) {
+        for (const [apiName, fieldSet] of Object.entries(fieldSets)) {
+            this.configureFieldSet(apiName, fieldSet);
+        }
+    }
+
+    getCurrentFieldSet() {
+        return this.fieldSets[this.currentFieldSetName];
+    }
+
+    isFieldInCurrentFieldSet(fieldApiName) {
+        return this.isFieldInFieldSet(this.getCurrentFieldSet(), fieldApiName);
+    }
+
+    isFieldInFieldSet(fieldSet, fieldApiName) {
+        let field = fieldSet.find(member => member.apiName === fieldApiName);
+
+        return field !== undefined;
+    }
+
+    hasContactField(fieldSet) {
+        let contactField = fieldSet.find(
+            member => member.apiName === CONTACT_FIELD.fieldApiName
+        );
+
+        return contactField !== undefined;
+    }
+
+    hasProgramEngagementField(fieldSet) {
+        let programEngagementField = fieldSet.find(
+            member => member.apiName === PROGRAM_ENGAGEMENT_FIELD.fieldApiName
+        );
+
+        return programEngagementField !== undefined;
+    }
+
+    configureFieldSet(apiName, fieldSet) {
+        let configuredFieldSet = [];
+
+        fieldSet.forEach(member => {
+            let field = { ...member };
+            // Number fields are size 1
+            // Program Engagment lookup is size 4
+            // Client lookup is size 3
+            // Everything else is size 2
+            // This means that the field set we ship with is exactly 12 wide
+            this.setSize(field);
+            this.setDefaultAttributes(fieldSet, field);
+            configuredFieldSet.push(field);
+        });
+        this.fieldSets[apiName] = configuredFieldSet;
+    }
+
+    setSize(field) {
+        if (SHORT_DATA_TYPES.includes(field.type)) {
+            field.size = 1;
+        } else if (field.apiName === PROGRAM_ENGAGEMENT_FIELD.fieldApiName) {
+            field.size = 4;
+        } else if (
+            field.apiName === CONTACT_FIELD.fieldApiName ||
+            LONG_DATA_TYPES.includes(field.type)
+        ) {
+            field.size = 3;
+        } else {
+            field.size = 2;
+        }
+    }
+
+    setDefaultAttributes(fieldSet, field) {
+        let hasContactField = this.isFieldInFieldSet(
```

Comment:
Did you mean to refactor these to use the boolean getters above?

Reply:
No, I think we definitely have room for improvement because we have this check in both places but I couldn't find a real smooth way to do it. Since we are doing this for each field set not just the current I didn't store it.